objects/src/accounts/account_id.rs

```diff
@@ -134,336 +129,471 @@ impl FromStr for AccountStorageMode {
 }
 
-impl From<AccountId> for u64 {
+impl From<AccountId> for u128 {
     fn from(id: AccountId) -> Self {
-        id.0.as_int()
+        let mut le_bytes = [0_u8; 16];
+        le_bytes[..8].copy_from_slice(&id.second_felt().as_int().to_le_bytes());
+        le_bytes[8..].copy_from_slice(&id.first_felt().as_int().to_le_bytes());
+        u128::from_le_bytes(le_bytes)
     }
 }
 
 /// Account IDs are used as indexes in the account database, which is a tree of depth 64.
 impl From<AccountId> for LeafIndex<ACCOUNT_TREE_DEPTH> {
     fn from(id: AccountId) -> Self {
-        LeafIndex::new_max_depth(id.0.as_int())
+        LeafIndex::new_max_depth(id.first_felt().as_int())
     }
 }
 
 // CONVERSIONS TO ACCOUNT ID
 // ================================================================================================
 
-/// Returns an [AccountId] instantiated with the provided field element.
-///
-/// # Errors
-/// Returns an error if:
-/// - If there are fewer than [AccountId::MIN_ACCOUNT_ONES] in the provided value.
-/// - If the provided value contains invalid account ID metadata (i.e., the first 4 bits).
-pub fn account_id_from_felt(value: Felt) -> Result<AccountId, AccountError> {
-    let int_value = value.as_int();
+impl TryFrom<[Felt; 2]> for AccountId {
+    type Error = AccountError;
 
-    let count = int_value.count_ones();
-    if count < AccountId::MIN_ACCOUNT_ONES {
-        return Err(AccountError::AccountIdTooFewOnes(count));
+    /// Returns an [`AccountId`] instantiated with the provided field elements where `elements[0]`
+    /// is taken as the first felt and `elements[1]` is taken as the second element.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if any of the ID constraints are not met. See the type documentation for
+    /// details.
+    fn try_from(elements: [Felt; 2]) -> Result<Self, Self::Error> {
+        account_id_from_felts(elements)
     }
-
-    let bits = (int_value & ACCOUNT_STORAGE_MASK) >> ACCOUNT_STORAGE_MASK_SHIFT;
-    match bits {
-        PUBLIC | PRIVATE => (),
-        _ => return Err(AccountError::InvalidAccountStorageMode(format!("0b{bits:b}"))),
-    };
-
-    Ok(AccountId(value))
 }
 
-impl TryFrom<Felt> for AccountId {
+impl TryFrom<[u8; 15]> for AccountId {
     type Error = AccountError;
 
-    /// Returns an [AccountId] instantiated with the provided field element.
+    /// Tries to convert a byte array in little-endian order to an [`AccountId`].
     ///
     /// # Errors
-    /// Returns an error if:
-    /// - If there are fewer than [AccountId::MIN_ACCOUNT_ONES] in the provided value.
-    /// - If the provided value contains invalid account ID metadata (i.e., the first 4 bits).
-    fn try_from(value: Felt) -> Result<Self, Self::Error> {
-        account_id_from_felt(value)
-    }
-}
+    ///
+    /// Returns an error if any of the ID constraints are not met. See the type documentation for
+    /// details.
+    fn try_from(bytes: [u8; 15]) -> Result<Self, Self::Error> {
```

Comment:
Yeah, I'm not seeing a clean solution here yet.

1. Agree that the current approach (with peeking into the first byte) is cleaner.
2. I don't think moving metadata to the most significant bits of the ID is a good solution as this will affect how accounts are distributed in the account tree.

> And as mentioned above, the `u128::to_be_bytes` and the `<[u8; 15]>` representation will never match 100% but if we have BE encoding then we only have to append a single zero byte to make them equivalent, which seems easy enough.

I think having this property is important - but the question is if it is worth giving up point 1 above for it. Maybe we can leave this to a follow-up issue (i.e., make `u128::to_be_bytes` and the `<[u8; 15]>` encodings consistent).